common/app/templates/headerInlineJS/bootSystemJS.scala.js

```diff
@@ -3,7 +3,8 @@
 @import conf.Static
 @import conf.Configuration
 
-@JavaScript(Static.js.systemJsSetupFragment)
+// Service worker only caches the bundles, so the offline page can only ask for those
+@JavaScript(templates.js.systemJsSetup(useBundlesInDev = item.id == "offline-page").body)
```

Comment:
This still seems really odd. Maybe another way to do this is to avoid using `main.scala.html` entirely. For one thing, a number of those fragments really won't work offline anyway. Then you can configure the bundles, etc however you like.